app/helpers/display_ad_helper.rb

```diff
@@ -3,7 +3,15 @@ def display_ads_placement_area_options_array
     DisplayAd::ALLOWED_PLACEMENT_AREAS_HUMAN_READABLE.zip(DisplayAd::ALLOWED_PLACEMENT_AREAS)
   end
 
-  def audience_segments_for_display_ads
-    AudienceSegment.human_readable_segments
+  def automatic_audience_segments_options_array
+    AudienceSegment.not_manual.pluck(:id, :type_of)
+      .map { |(id, type)| [AudienceSegment.human_readable_description_for(type), id] }
+  end
+
+  def single_audience_segment_option(billboard)
+    segment = billboard.audience_segment
+    raise if segment.blank?
```

Comment:
Let's raise a specific exception?

Reply:
Corrected, thanks! Also added a comment so the expectation is clear